src/mongo/client/dbclient.cpp

```diff
@@ -1471,6 +1471,7 @@ list<string> DBClientWithCommands::getDatabaseNames() {
 
 list<string> DBClientWithCommands::getCollectionNames(const string& db, const BSONObj& filter) {
     auto_ptr<DBClientCursor> infos = enumerateCollections(db, filter);
+    uassert(0, "socket error for mapping query", infos.get());
```

Comment:
@jrassi Yeah but in those instances the things were actually queries! I mean, I guess these are too, but I'm sure you can craft a more appropriate error message for here and below.